tools/logger/convert.c

```diff
@@ -1016,10 +1020,12 @@ static int dump_ldc_info(void)
 
 	while (remaining > 0) {
 		name = format_uid_raw(&uid_ptr[cnt], 0, 0, false, false);
+		if(!name)
+			name = "-"
 		uid_addr = get_uuid_key(&uid_ptr[cnt]);
 		fprintf(out_fd, "\t%p  %s\n", (void *)uid_addr, name);
 
-		if (name) {
+		if (name && name != "-") {
```

Comment:
You're making a value special without it needing to be special. I don't agree at all with this specific change.

Reply:
Ok, so please propose an alternative solution. What do we do if name is NULL?
Skip printing at all?

Reply:
Probably the only reason `name` could be NULL is because some `malloc` failed. In that very unlikely case the program should just stop and not print garbage. Please use some kind of `assert` either here or probably better inside `log_vasprintf()` (in case KW can understand that)